src/mlte/_private/job.py

```diff
@@ -0,0 +1,56 @@
+import sys
+import threading
+import subprocess
+from typing import List
+from pathlib import Path
+
+
+def _get_interpreter_path() -> Path:
+    """
+    Get the path to the current interpreter.
+
+    :return: Path to the Python executable currently being used.
+    :rtype: Path    
+    """
+    return Path(sys.executable)
+
+
+def spawn_python_job(script_path: str, arguments: List[str]) -> int:
+    """
+    Spawn the Python job from the given script and arguments, and return its process identifier.
+
+    :param script: The full path to a Python script with the training or equivalent process to run.
+    :type script: str
+
+    :param arguments: A list of string arguments for the process.
+    :type arguments: List[str[]
+
+    :return: Process id of the process that was created.
+    :rtype: int    
+    """
+    python_executable = _get_interpreter_path()
+    new_arguments = [script_path]
```

Comment:
Can use

```python
new_arguments = [script_path, *arguments]
```

to avoid initialize-then-modify antipattern.

Reply:
Sure. Your linter seems to be more strict that what I am using. Could you remind me what you are using to check, so I can incorporate this into my flow, and detect these things earlier?

Reply:
No worries, found them.